gluecodium/src/main/java/com/here/gluecodium/generator/dart/DartGeneratorPredicates.kt

```diff
@@ -44,6 +45,18 @@ internal class DartGeneratorPredicates(
             "allFieldsCtorIsPublic" to { limeStruct: Any ->
                 limeStruct is LimeStruct && allFieldsCtorIsPublic(limeStruct)
             },
+            "isInternal" to { element: Any ->
+                when (element) {
+                    // Dart has no type nesting, so all types are "outside" and have to check for an internal outer type.
+                    is LimeType -> {
+                        generateSequence(element) {
+                            limeReferenceMap[it.path.parent.toString()] as? LimeType
+                        }.any { CommonGeneratorPredicates.isInternal(it, DART) }
+                    }
+                    is LimeNamedElement -> CommonGeneratorPredicates.isInternal(element, DART)
+                    else -> throw GluecodiumExecutionException("Unsupported element type ${element.javaClass.name}")
```

Comment:
I preserved the original logic -- this predicate had been moved from `DartVisibilityResolver.kt`. I did not want to alter the way it works.

Reply:
This predicate was used in a single place, and now it's used instead of `CommonGeneratorPredicates.isInternal`, which does not throw. 
If we sure, that if new call site restricts what it passes to the predicate, then i'm ok with this.

Reply:
At the moment the predicate is being called only from two places:
1. `DartGenerator.kt --> generateDart()` -- this place applies the predicate to each element of `List<LimeType>`. Therefore, there is no chance to have an invalid type.
2. `DartVisibilityResolver.kt` -- that's the place from where the logic was moved to `DartGeneratorPredicates.kt`. It accepts `Any`. The logic has not been changed.